pallets/staking/src/lib.rs

```diff
@@ -0,0 +1,923 @@
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Origin to initialize staking
+		type AuthorityOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// Identifier for the class of asset.
+		type AssetId: Member
+			+ Parameter
+			+ Default
+			+ Copy
+			+ HasCompact
+			+ MaybeSerializeDeserialize
+			+ MaxEncodedLen
+			+ TypeInfo;
+
+		/// Multi currency mechanism.
+		type Currency: MultiLockableCurrency<Self::AccountId, CurrencyId = Self::AssetId, Balance = Balance>;
+
+		/// Staking period length in blocks.
+		#[pallet::constant]
+		type PeriodLength: Get<Self::BlockNumber>;
+
+		/// Pallet id.
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+
+		/// HDX Asset ID
+		#[pallet::constant]
+		type HdxAssetId: Get<Self::AssetId>;
+
+		/// Min amount user must stake.
+		#[pallet::constant]
+		type MinStake: Get<Balance>;
+
+		/// Weight of the time points in total points calculations.
+		#[pallet::constant]
+		type TimePointsWeight: Get<Permill>;
+
+		/// Weight of the action points in total points calculations.
+		#[pallet::constant]
+		type ActionPointsWeight: Get<Permill>;
+
+		/// Number of time points users receive for each period.
+		#[pallet::constant]
+		type TimePointsPerPeriod: Get<u8>;
+
+		/// Number of periods user can't claim rewards for. User can exit but won't receive any rewards.
+		/// If he stay longer than `UnclaimablePeriods` he will receive rewards also for these periods.
+		#[pallet::constant]
+		type UnclaimablePeriods: Get<Period>;
+
+		/// Weight of the actual stake in slash points calculation. Bigger the value lower the calculated slash points.
+		#[pallet::constant]
+		type CurrentStakeWeight: Get<u8>;
+
+		/// Function returning percentage of rewards to pay based on number of points user
+		/// accumulated.
+		type PayablePercentage: PayablePercentage<Point>;
+
+		/// The block number provider.
+		type BlockNumberProvider: BlockNumberProvider<BlockNumber = Self::BlockNumber>;
+
+		/// Position identifier type.
+		type PositionItemId: Member + Parameter + Default + Copy + HasCompact + AtLeast32BitUnsigned + MaxEncodedLen;
+
+		/// Collection id type
+		type CollectionId: TypeInfo + MaxEncodedLen;
+
+		/// NFT collection id
+		#[pallet::constant]
+		type NFTCollectionId: Get<Self::CollectionId>;
+
+		/// Provides ability to freeze a collection.
+		type Collections: Freeze<Self::AccountId, Self::CollectionId>;
+
+		/// Non fungible handling - mint, burn, check owner
+		type NFTHandler: Mutate<Self::AccountId>
+			+ Create<Self::AccountId>
+			+ Inspect<Self::AccountId, ItemId = Self::PositionItemId, CollectionId = Self::CollectionId>
+			+ InspectEnumerable<Self::AccountId, ItemId = Self::PositionItemId, CollectionId = Self::CollectionId>;
+
+		/// Max amount of votes the user can have at any time.
+		#[pallet::constant]
+		type MaxVotes: Get<u32>;
```

Comment:
We are not arranging it that way in other pallets. If we want to do stuff like this we should create "style guide" first otherwise it depends only on preferences of the reviewer.

Reply:
what is the reason for not arranging the constants next to each other?

Reply:
I'm happy to do it if we'll decide to do it. Otherwise it's just preference of one person vs another.  It won't be consistent through all the pallets anyway so I don't see benefit of investing time into it

Reply:
Arranging config parameters improves readability, mainly for pallets with a lot of parameters. I believe it belongs under common best practices, hence there is no need for a style guide. I don't see a reason why not to rearrange it, especially when a reviewer points it out. I would argue that putting constants next to each other makes sense, so it's not a matter of preference. Just my 2 cents :v:

Reply:
ok, I moved it